std-bits/table/src/main/java/org/enso/table/data/column/operation/map/bool/BooleanIsInOp.java

```diff
@@ -0,0 +1,65 @@
+package org.enso.table.data.column.operation.map.bool;
+
+import java.util.BitSet;
+import java.util.List;
+import org.enso.table.data.column.operation.map.MapOperation;
+import org.enso.table.data.column.storage.BoolStorage;
+import org.enso.table.data.column.storage.Storage;
+
+/**
+ * A specialized implementation for the IS_IN operation on booleans - since booleans have just three
+ * possible values we can have a highly efficient implementation that does not even rely on hashmap
+ * and after processing the input vector, performs the checks in constant time.
+ */
+public class BooleanIsInOp extends MapOperation<BoolStorage> {
+  public BooleanIsInOp() {
+    super(Storage.Maps.IS_IN);
+  }
+
+  @Override
+  public Storage runMap(BoolStorage storage, Object arg) {
+    if (arg instanceof List) {
+      return runMap(storage, (List<?>) arg);
+    } else {
+      throw new IllegalArgumentException("Argument to `is_in` must be a vector.");
+    }
+  }
+
+  public Storage runMap(BoolStorage storage, List<?> arg) {
+    boolean hadTrue = false;
+    boolean hadFalse = false;
+    boolean hadNull = false;
+
+    for (Object o : arg) {
```

Comment:
You are right, and I can do that too for `hadNull && !hadTrue && !hadTrue` and many other cases too!

The code became a bit more complicated, but for half cases it became O(1) and for other cases it uses much faster operations (operating on words - so groups of I think 64 bits at once) than iteration. Overall the result should be significantly faster, but I don't think we want to spend time measuring this.

Since the code is much more complicated it's also more error prone. I tried adding comments clarifying my intents, but still managed to have bugs in the first iteration - I added a much more robust test suite which basically tests every combination so that we can be assured that the new solution is correct.